db/db_impl/db_impl.cc

```diff
@@ -2315,6 +2326,21 @@ Status DBImpl::MultiGetImpl(
       super_version->current->MultiGet(read_options, &range, callback,
                                        is_blob_index);
     }
+    curr_value_size = range.GetValueSize();
+    if (curr_value_size > read_options.value_size_soft_limit) {
+      break;
+    }
+  }
+
+  // Set all remaining keys status to abort
+  if (keys_left && s != Status::TimedOut() &&
```

Comment:
We can just combine this block with the one in lines 2357-2364 as the logic is the same.